### PR TITLE
Fix KSTO

### LINKED ISSUE #1376: KSTO view crashes on iOS
To reproduce: open the Ksto view on iOS, wait about 10 seconds.

Something in here is requesting data over the network, as I can see the network spinner spinning.

### PR DESCRIPTION
**Pull request by drewvolz**
Closes #1376 

* Forked react-native-video to fix iOS crash
* Lib function to timeout on a promise (thanks @hawkrives)
* Check to see if m3u8 chunklist is returned within 6 seconds
* Pause stream state if we do not see data coming in from stream
* Run this check every time we unpause the stream

---

Edit from  @hawkrives:

I've added a simple state machine to this component.

The Button can be in one of three states: playing, paused, or checking. It can only ever be in one of those states, which simplifies the logic around checking the status.

Screenshots:

On Start | Checking | Playing
--- | --- | --- 
screen shot 2017-09-16 at 12 21 23 pm | screen shot 2017-09-16 at 12 21 25 pm |  screen shot 2017-09-16 at 12 21 26 pm

### CODE REVIEW

File: source/views/streaming/radio.js [hunk 1]
```diff
@@ -34,31 +36,61 @@ export default class KSTOView extends React.PureComponent {
...
+    if (this.state.uplinkError) {
+      this.setState(() => ({paused: true}))
+    }
+  }
+
   changeControl = () => {
     this.setState(state => ({paused: !state.paused}))
+
+    // If we try to play...
+    if (this.state.paused) {
```
Comment (hawkrives) [inline source/views/streaming/radio.js]: This is not guaranteed to have the new state. I almost think that this block should go into the setState callback.

File: source/views/streaming/radio.js [hunk 2]
```diff
@@ -34,31 +36,61 @@ export default class KSTOView extends React.PureComponent {
...
+      this.setState(() => ({uplinkError: false, message: ''}))
+    } catch (err) {
+      this.setState(() => ({
+        uplinkError: true,
+        message: 'The KSTO stream is down. Sorry!',
+      }))
+    }
+
+    // If the stream is down or we had an error, pause the player
+    if (this.state.uplinkError) {
```
Comment (hawkrives) [inline source/views/streaming/radio.js]: UplinkError is not guaranteed to have been set by this point.

File: source/views/streaming/radio.js (hunk 1 above)
Comment (hawkrives) [reply source/views/streaming/radio.js]: I think I know what I'd like this to do; let me take a stab at it in a bit.

**Comment by hawkrives** [comment]
oops.

**Comment by hawkrives** [comment]
fixed?

**Review by rye — APPROVED** [review]

**Comment by hawkrives** [comment]
I'm still plotting to work on this before merging, fyi.

**Review by hawkrives — APPROVED** [review]
I shall now approve my own work lol

**Review by rye — APPROVED** [review]

**Comment by drewvolz** [comment]
Your state machine is much nicer. Good work!

**Comment by hawkrives** [comment]
Oh munglfraijs.

Fixing prettier.

**Pull request merged**